### PR TITLE
Add UserId parameter to AccessValidator

### PR DESCRIPTION
**Pull request by KostinP**
Motivation: if there is no context, then you can pass the UserId, and not create it from scratch

### CODE REVIEW

**Review by 24eeeee — APPROVED** [review]

**Review by MariyaSin — APPROVED** [review]

**Review by TIove — APPROVED** [review]

File: src/Kernel/AccessValidatorEngine/AccessValidator.cs [hunk 1]
```diff
@@ -33,14 +33,21 @@ public class AccessValidator : IAccessValidator
         }
 
         /// <inheritdoc/>
-        public bool HasRights(params int[] rightIds)
+        public bool HasRights(Guid? userId, params int[] rightIds)
```
Comment (DarkFelas) [inline src/Kernel/AccessValidatorEngine/AccessValidator.cs:40]: not good. plz add new method HasRights

**Review by DarkFelas — CHANGES_REQUESTED** [review]
check comment

**Review by borovHokage — APPROVED** [review]

File: LT.DigitalOffice.Kernel.nuspec [hunk 2]
```diff
@@ -2,17 +2,18 @@
 <package >
   <metadata>
     <id>LT.DigitalOffice.Kernel</id>
-    <version>1.1.22</version>
+    <version>1.1.23</version>
```
Comment (DarkFelas) [inline LT.DigitalOffice.Kernel.nuspec]: we already have published 1.1.22?

**Review by DarkFelas — CHANGES_REQUESTED** [review]

File: LT.DigitalOffice.Kernel.nuspec (hunk 2 above)
Comment (KostinP) [reply LT.DigitalOffice.Kernel.nuspec]: Fixed.

File: src/Kernel/AccessValidatorEngine/AccessValidator.cs (hunk 1 above)
Comment (KostinP) [reply src/Kernel/AccessValidatorEngine/AccessValidator.cs:40]: Fixed good?

File: src/Kernel/AccessValidatorEngine/AccessValidator.cs [hunk 3]
```diff
@@ -55,12 +59,31 @@ public bool HasRights(params int[] rightIds)
...
 
         /// <inheritdoc/>
-        public bool IsAdmin()
+        public bool HasRights(params int[] rightIds)
+        {
+            return CheckUserRights(null, rightIds);
+        }
+
+        /// <inheritdoc/>
+        public bool HasRights(Guid? userId, params int[] rightIds)
```
Comment (DarkFelas) [inline src/Kernel/AccessValidatorEngine/AccessValidator.cs]: what for this method?

File: src/Kernel/AccessValidatorEngine/AccessValidator.cs [hunk 4]
```diff
@@ -55,12 +59,31 @@ public bool HasRights(params int[] rightIds)
...
+        {
+            return CheckUserRights(userId, rightIds);
+        }
+
+        /// <inheritdoc/>
+        public bool IsAdmin(Guid? userId = null)
         {
-            _userId = _httpContext.GetUserId();
+            if (userId == null)
```
Comment (DarkFelas) [inline src/Kernel/AccessValidatorEngine/AccessValidator.cs]: if (!userId.HasValue)
{
 userId = _httpContext.GetUserId();
}

File: src/Kernel/AccessValidatorEngine/AccessValidator.cs [hunk 5]
```diff
@@ -55,12 +59,31 @@ public bool HasRights(params int[] rightIds)
...
+            {
+                userId = _httpContext.GetUserId();
+            }
+            else
```
Comment (DarkFelas) [inline src/Kernel/AccessValidatorEngine/AccessValidator.cs]: remove

File: src/Kernel/AccessValidatorEngine/AccessValidator.cs [hunk 6]
```diff
@@ -55,12 +59,31 @@ public bool HasRights(params int[] rightIds)
...
+            {
+                userId = (Guid)userId;
+            }
 
             Response<IOperationResult<bool>> result = _requestClientUS.GetResponse<IOperationResult<bool>>(
-                ICheckUserIsAdminRequest.CreateObj(_userId),
+                ICheckUserIsAdminRequest.CreateObj((Guid)userId),
```
Comment (DarkFelas) [inline src/Kernel/AccessValidatorEngine/AccessValidator.cs]: userId.Value

File: src/Kernel/AccessValidatorEngine/AccessValidator.cs [hunk 7]
```diff
@@ -32,18 +30,24 @@ public class AccessValidator : IAccessValidator
...
+                userId = _httpContext.GetUserId();
+            }
+            else
+            {
+                userId = (Guid)userId;
+            }
 
             Response<IOperationResult<bool>> result = _requestClientCR.GetResponse<IOperationResult<bool>>(
-                ICheckUserRightsRequest.CreateObj(_userId, rightIds),
+                ICheckUserRightsRequest.CreateObj((Guid)userId, rightIds),
```
Comment (DarkFelas) [inline src/Kernel/AccessValidatorEngine/AccessValidator.cs]: userId.Value

File: src/Kernel/AccessValidatorEngine/AccessValidator.cs [hunk 8]
```diff
@@ -32,18 +30,24 @@ public class AccessValidator : IAccessValidator
...
             {
                 throw new ArgumentException("Can not check empty rights array.", nameof(rightIds));
             }
 
-            _userId = _httpContext.GetUserId();
+            if (userId == null)
+            {
+                userId = _httpContext.GetUserId();
+            }
+            else
```
Comment (DarkFelas) [inline src/Kernel/AccessValidatorEngine/AccessValidator.cs]: remove

File: src/Kernel/AccessValidatorEngine/AccessValidator.cs [hunk 9]
```diff
@@ -32,18 +30,24 @@ public class AccessValidator : IAccessValidator
...
-        public bool HasRights(params int[] rightIds)
+        private bool CheckUserRights(Guid? userId, int[] rightIds)
         {
             if (rightIds == null || !rightIds.Any())
             {
                 throw new ArgumentException("Can not check empty rights array.", nameof(rightIds));
             }
 
-            _userId = _httpContext.GetUserId();
+            if (userId == null)
```
Comment (DarkFelas) [inline src/Kernel/AccessValidatorEngine/AccessValidator.cs]: if (!userId.HasValue)

File: src/Kernel/AccessValidatorEngine/AccessValidator.cs [hunk 10]
```diff
@@ -32,18 +30,24 @@ public class AccessValidator : IAccessValidator
             _httpContext = httpContextAccessor.HttpContext;
         }
 
-        /// <inheritdoc/>
-        public bool HasRights(params int[] rightIds)
+        private bool CheckUserRights(Guid? userId, int[] rightIds)
```
Comment (DarkFelas) [inline src/Kernel/AccessValidatorEngine/AccessValidator.cs]: why private?

**Review by DarkFelas — CHANGES_REQUESTED** [review]

File: src/Kernel/AccessValidatorEngine/AccessValidator.cs (hunk 10 above)
Comment (KostinP) [reply src/Kernel/AccessValidatorEngine/AccessValidator.cs]: Fixed.

File: src/Kernel/AccessValidatorEngine/AccessValidator.cs (hunk 9 above)
Comment (KostinP) [reply src/Kernel/AccessValidatorEngine/AccessValidator.cs]: Fixed.

File: src/Kernel/AccessValidatorEngine/AccessValidator.cs (hunk 8 above)
Comment (KostinP) [reply src/Kernel/AccessValidatorEngine/AccessValidator.cs]: Fixed.

File: src/Kernel/AccessValidatorEngine/AccessValidator.cs (hunk 7 above)
Comment (KostinP) [reply src/Kernel/AccessValidatorEngine/AccessValidator.cs]: Fixed.

File: src/Kernel/AccessValidatorEngine/AccessValidator.cs (hunk 6 above)
Comment (KostinP) [reply src/Kernel/AccessValidatorEngine/AccessValidator.cs]: Fixed.

File: src/Kernel/AccessValidatorEngine/AccessValidator.cs (hunk 5 above)
Comment (KostinP) [reply src/Kernel/AccessValidatorEngine/AccessValidator.cs]: Fixed.

File: src/Kernel/AccessValidatorEngine/AccessValidator.cs (hunk 4 above)
Comment (KostinP) [reply src/Kernel/AccessValidatorEngine/AccessValidator.cs]: Fixed.

File: src/Kernel/AccessValidatorEngine/AccessValidator.cs (hunk 3 above)
Comment (KostinP) [reply src/Kernel/AccessValidatorEngine/AccessValidator.cs]: Fixed.

**Review by DarkFelas — APPROVED** [review]

**Pull request merged**